### ISSUE TITLE
NameError: name 'is_allowed_path' is not defined

### ISSUE THREAD

**Issue by webcharlyx** [letorbi/synoindexwatcher#47]
latest version of your script installed successfully:
```
sudo python3 -m pip install --upgrade synoindexwatcher
Requirement already satisfied: synoindexwatcher in /volume1/@appstore/py3k/usr/local/lib/python3.8/site-packages (0.11.4)
Requirement already satisfied: inotifyrecursive>=0.3.5 in /volume1/@appstore/py3k/usr/local/lib/python3.8/site-packages (from synoindexwatcher) (0.3.5)
Requirement already satisfied: inotify-simple>=1.3.5 in /volume1/@appstore/py3k/usr/local/lib/python3.8/site-packages (from inotifyrecursive>=0.3.5->synoindexwatcher) (1.3.5)
```
doing `sudo python3 -m synoindexwatcher --rebuild-index --config=/usr/local/etc/synoindexwatcher.conf --loglevel=DEBUG` results in:
```
2021-04-10 00:06:46,048 ERROR An uncaught exception occurred
Traceback (most recent call last):
  File "/var/packages/py3k/target/usr/local/lib/python3.8/site-packages/synoindexwatcher/__main__.py", line 218, in <module>
    start()
  File "/var/packages/py3k/target/usr/local/lib/python3.8/site-packages/synoindexwatcher/__main__.py", line 177, in start
    add_to_index_recursive(path)
  File "/var/packages/py3k/target/usr/local/lib/python3.8/site-packages/synoindexwatcher/__main__.py", line 62, in add_to_index_recursive
    __add_to_index_recursive(path, name)
  File "/var/packages/py3k/target/usr/local/lib/python3.8/site-packages/synoindexwatcher/__main__.py", line 48, in __add_to_index_recursive
    if not is_allowed_path(name, -1, is_dir):
NameError: name 'is_allowed_path' is not defined
```
using Python 3.8.6-0154 on DS418play

seems to fail since python got update by package center...

**Comment by letorbi** [comment]
Thanks for the bug report. All references to `is_allowed_path()` should have been replaced by `filter()` a long time ago...

Could you apply the following patch to Synoindex Watcher's main module on you system and check whether it fixes the bug?

```diff
diff --git a/synoindexwatcher/__main__.py b/synoindexwatcher/__main__.py
index b56c853..0997530 100644
--- a/synoindexwatcher/__main__.py
+++ b/synoindexwatcher/__main__.py
@@ -42,10 +42,10 @@ else:
     def __fix_encoding(s):
         return s
 
-def __add_to_index_recursive(path, name):
+def __add_to_index_recursive(path, name, filter):
     fullpath = os.path.join(path, name)
     is_dir = os.path.isdir(fullpath) 
-    if not is_allowed_path(name, -1, is_dir):
+    if not filter(name, path, is_dir):
         logging.debug("Skipping path %s" % fullpath)
         return
     add_to_index(fullpath, is_dir)
@@ -57,9 +57,9 @@ def call_command(args):
     logging.debug("Calling '%s'" % " ".join(args))
     return subprocess.check_output(args, stderr=subprocess.STDOUT)
 
-def add_to_index_recursive(fullpath):
+def add_to_index_recursive(fullpath, filter):
     (path, name) = os.path.split(fullpath)
-    __add_to_index_recursive(path, name)
+    __add_to_index_recursive(path, name, filter)
 
 def add_to_index(fullpath, is_dir):
     if is_dir:
@@ -159,6 +159,7 @@ def on_sigterm(signal, frame):
 def start():
     config = read_config()
     args = parse_arguments(config)
+    filter = build_filter(args)
 
     if args.generate_init:
         print(files.generateInit(sys.argv))
@@ -174,14 +175,13 @@ def start():
     if args.rebuild_index:
         print("Adding files to media-index (this may take some time)...")
         for path in args.path:
-            add_to_index_recursive(path)
+            add_to_index_recursive(path, filter)
         return
 
     signal.signal(signal.SIGTERM, on_sigterm)
 
     try:
         inotify = INotify()
-        filter = build_filter(args)
         mask = flags.DELETE | flags.CREATE | flags.MOVED_TO | flags.MOVED_FROM | flags.MODIFY | flags.CLOSE_WRITE
         for path in args.path:
             logging.info("Adding watch for path: %s", path)
```

**Comment by petermarks** [comment]
I ran into this issue and fixed it before I saw your patch. I basically did the same, but also made the auxiliary function local so you don't need to pass the filter on every recursive step. I've submitted pr #54.

**Comment by letorbi** [comment]
I've merged PR #54. Thanks again, Peter :)
